src/imap.rs

```diff
@@ -1030,9 +1030,12 @@ impl Imap {
             .await?;
 
         self.prepare(context).await?;
-        self.select_folder(context, Some(folder)).await?;
 
         for (target, rowid_set, uid_set) in UidGrouper::from(rows) {
+            // Select folder inside the loop to avoid selecting it if there are no pending
+            // MOVE/DELETE operations.
+            self.select_folder(context, Some(folder)).await?;
+
```

Comment:
BTW note that `select_folder()` doesn't select the folder if it's already selected, so this doesn't increase the number of run SELECT statements.